frontend/src/features/leases/detail/LeasePages/payment/table/columns.tsx

```diff
@@ -0,0 +1,277 @@
+import { Button } from 'components/common/form';
+import { InlineFlexDiv } from 'components/common/styles';
+import TooltipIcon from 'components/common/TooltipIcon';
+import {
+  ColumnWithProps,
+  renderBooleanAsYesNo,
+  renderMoney,
+  renderTypeCode,
+} from 'components/Table';
+import { ILeasePayment } from 'interfaces';
+import { IFormLeaseTerm, ILeaseTerm } from 'interfaces/ILeaseTerm';
+import moment from 'moment';
+import { FaTrash } from 'react-icons/fa';
+import { MdEdit } from 'react-icons/md';
+import { CellProps } from 'react-table';
+import { ISystemConstant } from 'store/slices/systemConstants';
+import styled from 'styled-components';
+import { formatMoney, prettyFormatDate } from 'utils';
+
+function initialOrRenewalTerm({ row: { original, index } }: CellProps<ILeaseTerm, string>) {
+  return index === 0 ? 'Initial term' : `Renewal ${index}`;
+}
+
+function startAndEndDate({ row: { original, index } }: CellProps<ILeaseTerm, string>) {
+  return `${prettyFormatDate(original.startDate)} - ${prettyFormatDate(original.expiryDate)}`;
+}
+
+const renderGst = (gstDecimal?: number) =>
+  function({ row: { original, index } }: CellProps<ILeaseTerm, string>) {
+    gstDecimal = original.isGstEligible ? gstDecimal : undefined;
+    return gstDecimal !== undefined && original.paymentAmount !== undefined
+      ? formatMoney(calculateExpectedGst(original.paymentAmount, gstDecimal))
+      : '-';
+  };
+
+const renderExpectedTotal = (gstDecimal?: number) =>
+  function({ row: { original, index } }: CellProps<ILeaseTerm, string>) {
+    gstDecimal = original.isGstEligible ? gstDecimal : 0;
+    return gstDecimal !== undefined && original.paymentAmount !== undefined
+      ? formatMoney(calculateExpectedTotal(original.paymentAmount, gstDecimal))
+      : '-';
+  };
+
+function renderActualTotal({ row: { original, index } }: CellProps<ILeaseTerm, string>) {
+  const total = formatMoney(
+    (original.payments ?? []).reduce((sum: number, p: ILeasePayment) => (sum += p.amountTotal), 0),
+  );
+  return original.isTermExercised ? total : '-';
+}
+
+const renderExpectedTerm = (gstDecimal?: number) =>
+  function({ row: { original, index } }: CellProps<ILeaseTerm, string>) {
+    gstDecimal = original.isGstEligible ? gstDecimal : undefined;
+    if (!original.startDate || !original.expiryDate || original.paymentAmount === undefined) {
+      return '-';
+    }
+    const expectedTerm = calculateExpectedTermAmount(
+      original.leasePmtFreqTypeCode?.description ?? '',
+      original.startDate,
+      original.expiryDate,
+      original.paymentAmount,
+      gstDecimal ?? 0,
+    );
+    return expectedTerm !== undefined ? formatMoney(expectedTerm) : '-';
+  };
+
+function calculateExpectedGst(amount: number, gst: number) {
+  return amount * (gst / 100);
```

Comment:
nit: gst/100 will always be the same value since gst is a constant.

Reply:
Not sure what you mean, gst can't be hardcoded since it comes from the DB. I could memo this but I don't think the overhead is important.

Reply:
not hardcoded, but once calculated it won't change.

Reply:
This would be more obvious if there is an object that contains the methods and members for the calculated information